avogadro/qtplugins/bonding/bonding.cpp

```diff
@@ -121,7 +130,30 @@ void Bonding::bond2()
 
 void Bonding::clearBonds()
 {
-  m_molecule->clearBonds();
+  // remove any bonds connected to the selected atoms
```

Comment:
The first behavior is a bug (fix coming). I have to think a bit about the UX for adding bonds back if only one atom is selected. To me, it makes sense that bonds to a selected atom should disappear. Does it make sense to re-bond if only one atom is selected?

To me, it made the most sense that you would need to select a bunch of atoms to bond only in that set.

Keep in mind that the bond add/remove is undoable.